greymatter-io/gitops-core
# Fault injection Filter

[sc-17696]
This allows the user to inject faults (aborts or delays) intentionally via header options or set percentages of requests.

To test:
1. Set `_enable_fault_injection: true` in `edge.cue`, then run `cue eval -c ./gm/outputs --out text -e mesh_configs_yaml` and ensure that the fault injection filter is set on the listener (if you have yq, you can do `| yq -o=json '. | select(.listener_key=="edge")| sort_keys(..)'` after the above command)
2. Set `_enable_fault_injection: false` and rerun the above command. The configuration should not be present in the listener.
3. To test the functionality of the filter, apply the filter to a running mesh with the defaults in `intermediates.cue`, and attempt to query the edge with `curl --head http://{EXTERNAL_IP}:10808 -H "x-envoy-fault-abort-request: 551"`. Approximately 50% of the requests will fail with the supplied status code. Additionally, 50% of all requests will experience a fixed delay of `5s` before returning.
